### DISCUSSION TITLE
LanguageDetector needs at least 2 languages to choose from Error

### DISCUSSION THREAD

**Discussion by JimLynchCodes** [pemistahl/lingua-rs#170] [Q&A]
Hi, I copied the sample code into my project and deleted all the languages except english:
```rust

use lingua::Language::{English};
use lingua::{Language, LanguageDetector, LanguageDetectorBuilder};

pub fn is_valid_english_word(word: String) {

    let languages = vec![English];
    let detector: LanguageDetector = LanguageDetectorBuilder::from_languages(&languages).build();
    let detected_language: Option<Language> = detector.detect_language_of("languages are awesome");

    detector.compute_language_confidence_values(text)

    assert_eq!(detected_language, Some(English));
}
```

My use case is that I am building a game like wordle, and I want some function that I can pass an arbitrary string into and have the function return a bool whether the string is a valid english word or not...

Is there a reason this is not supported by lingua-rs? Or am is there some other function I could be calling to efficiently accomplish this? thanks!

PS. here is the full error I see when running it:
```
thread 'main' panicked at 'LanguageDetector needs at least 2 languages to choose from', /Users/jim/.cargo/registry/src/github.com-1ecc6299db9ec823/lingua-1.4.0/src/builder.rs:91:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by serega** [comment]
I am not the author of the library, but I think I can answer this question. 
This library detects ( `predicts` ) the most likely language of a given text. If you initialize the `LanguageDetetor` with only single language, say English the the most likely language will be always be English because there are no other languages to choose from, which is why the detector requires at least two languages. 
The library does not validate whether a given text is grammatically correct or whether it contains valid words in any language. You probably need a list of English words for your use case.

**Comment by pemistahl** [comment]
@serega is right. Lingua needs to load at least two language models so that it can compare the ngram probabilities of those two languages in order to choose the most likely language. There is no absolute confidence metric yet which can tell you whether some text is English or not without looking at other languages for comparison. I will try to work on such a feature but that's pretty tough to implement.
